sdk/messaging/azservicebus/internal/errors.go

```diff
@@ -91,17 +91,18 @@ var amqpConditionsToRecoveryKind = map[amqp.ErrorCondition]recoveryKind{
 	amqp.ErrorCondition("com.microsoft:operation-cancelled"): RecoveryKindNone,
 
 	// Link recovery needed
-	amqp.ErrorDetachForced: RecoveryKindLink, // "amqp:link:detach-forced"
+	amqp.ErrorDetachForced:          RecoveryKindLink, // "amqp:link:detach-forced"
+	amqp.ErrorTransferLimitExceeded: RecoveryKindLink,
 
 	// Connection recovery needed
 	amqp.ErrorConnectionForced: RecoveryKindConn, // "amqp:connection:forced"
+	amqp.ErrorInternalError:    RecoveryKindConn, // "amqp:internal-error"
```

Comment:
Internal errors are always recoverable at the connection level?

Reply:
The main issue with InternalError is that the cause of it is not known on our end. So the safest thing to do (rather than getting stuck in an infinite loop) is to at least try to recreate the entire thing.

We do our best internally to not lose any state, so at worse it's hopefully just a performance disruption.